reference-service/src/main/java/com/transformuk/hee/tis/reference/service/api/LeavingReasonResource.java

```diff
@@ -194,38 +192,46 @@ public ResponseEntity<Void> deleteLeavingReason(@PathVariable Long id) {
   }
 
   /**
-   * POST /leaving-reasons/exist : check if leaving reasons exist.
+   * POST /leaving-reasons/match : check if there's a role in the database that matches the code
+   * provided, regardless of casing.
    *
    * @param codes             the codes of the leaving reasons to check
    * @param columnFilterJson  the column filters to apply
-   * @return a map contains the code as key and boolean value for its existence
-   * @throws IOException when getting column filters
+   * @return a map            Where a key is the code to be matched,
+   *     and a value is the code that was matched from the database.
+   * @throws IOException      when getting column filters
    */
-  @PostMapping("/leaving-reasons/exist")
-  public ResponseEntity<Map<String, Boolean>> leavingReasonsExist(
+  @PostMapping("/leaving-reasons/match")
+  public ResponseEntity<Map<String, String>> leavingReasonsMatch(
       @RequestBody List<String> codes,
       @RequestParam(value = "columnFilters", required = false) String columnFilterJson
   ) throws IOException {
     codes = codes.stream().map(code -> getConverter(code).decodeUrl().toString())
         .collect(Collectors.toList());
-    LOGGER.debug("REST request to check leaving reasons existence: {}", codes);
-    Specification<LeavingReason> specs = Specification.where(in("code", new ArrayList<>(codes)));
+    LOGGER.debug("REST request to check leaving reasons match: {}", codes);
+
 
     List<Class> filterEnumList = Lists.newArrayList(Status.class);
     List<ColumnFilter> columnFilters =
         ColumnFilterUtil.getColumnFilters(columnFilterJson, filterEnumList);
 
-    for (ColumnFilter columnFilter : columnFilters) {
-      specs = specs.and(in(columnFilter.getName(), columnFilter.getValues()));
-    }
-
-    List<LeavingReasonDto> leavingReasonDtos = service.findAll(null, columnFilters);
```

Comment:
There's a bug here. The specs are NOT used anywhere, and `findAll` method returns all leavingReasonDtos. This PR will fix it.